support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java

```diff
@@ -95,22 +101,57 @@ public SSOSamlIdPPostProfileHandlerEndpoint(final CasConfigurationProperties cas
      */
     @GetMapping(produces = MediaType.APPLICATION_XML_VALUE)
     @ResponseBody
-    public ResponseEntity<Object> produce(final HttpServletRequest request, final HttpServletResponse response) {
+    public ResponseEntity<Object> produceGet(final HttpServletRequest request, final HttpServletResponse response) {
+
+        val username = request.getParameter("username");
+        val password = request.getParameter("password");
+        val entityId = request.getParameter(SamlProtocolConstants.PARAMETER_ENTITY_ID);
+        val encrypt = Boolean.parseBoolean(request.getParameter("encrypt"));
+        return produce(request, response, username, password, entityId, encrypt);
+    }
+
+    /**
+     * Produce response entity.
+     *
+     * @param request  the request
+     * @param response the response
+     * @param map the RequestBody
+     * @return the response entity
+     */
+    @PostMapping(produces = MediaType.APPLICATION_XML_VALUE)
+    @ResponseBody
+    public ResponseEntity<Object> producePost(final HttpServletRequest request,
+                                              final HttpServletResponse response,
+                                              final @RequestBody Map<String, String> map) {
+
+        val username = map.get("username");
```

Comment:
Not sure I follow this; if you mean to support both GET and POST, a RequestMapping on the original method with both methods should do the job, and would be less code.

Reply:
Hmm, How do you get parameters out of the RequestBody, compared to getting them as url params?

Reply:
I think the method signature should just be using @RequestParam, and then Spring will take care of it.

Reply:
I will give it a try

Reply:
yes, content header is application/json and sending the request body as a JSON string

Reply:
Most likely, the method here should indicated via "consumes" that it accepts JSON. That's likely where the bad-request comes from. Can you share a sample JSON payload with me?

Reply:
No worries, always good to explore all options.